doc/classes/ColorPicker.xml

```diff
@@ -82,6 +77,18 @@
 		</signal>
 	</signals>
 	<constants>
+		<constant name="MODE_RGB" value="0" enum="ColorModeType">
+			Allows editing the color with Red/Green/Blue sliders.
+		</constant>
+		<constant name="MODE_HSV" value="1" enum="ColorModeType">
+			Allows editing the color with Hue/Saturation/Value sliders.
+		</constant>
+		<constant name="MODE_RAW" value="2" enum="ColorModeType">
+			Allows the color R, G, B component values to go beyond 1.0, which can be used for certain special operations that require it (like tinting without darkening or rendering sprites in HDR).
+		</constant>
+		<constant name="MODE_OKHSL" value="3" enum="ColorModeType">
+			Allows editing the color with Hue/Saturation/Lightness sliders.
```

Comment:
This could include more details. It's a different color space from HSV and it's more designed for picking similar colors or something. @fire will know more.

Reply:
Source: https://bottosson.github.io/posts/colorpicker/

> Picking colors is a common operation in many applications and over the years color pickers have become fairly standardized. Ubiquitous today are color pickers based on [HSL and HSV](https://en.wikipedia.org/wiki/HSL_and_HSV). They are simple transformations of RGB values to alternative coordinates chosen to better correlate with perceptual qualities.

> Despite color picking playing a big role in a lot of applications, the design of color pickers isn’t a particularly well researched topic. While some variation exist in the widgets themselves, the choice of HSL or HSV is mostly taken for granted, with only a few exceptions.

> By far the most used color spaces today for color picking are [HSL and HSV](https://en.wikipedia.org/wiki/HSL_and_HSV), two representations introduced in the classic 1978 paper [“Color Spaces for Computer Graphics”](https://doi.org/10.1145/800248.807362). HSL and HSV designed to roughly correlate with perceptual color properties while being very simple and cheap to compute.

> Worth noting is that HSL and HSV are not quite color spaces on their own, they are transformations from a source RGB color space. For each set of RGB primaries and transfer functions, the transformation to HSL and HSV produces unique color spaces. Today HSL and HSV are most commonly used together with the sRGB color space, so that is what we will look at here and we will here use HSL and HSV to refer to HSL and HSV for the sRGB color space.

> Also useful to note is that HSL and HSV are not continuously differentiable, so that limits their use with numerical optimization and machine learning.

> One of the main advantages of HSL and HSV over the different Lab color spaces is that they map the sRGB gamut to a cylinder. This makes them easy to use since all parameters can be changed independently, without the risk of creating colors outside of the target gamut.

> We will attempt to do just that by creating new color spaces similar to HSL and HSV but that better match perception. This will be done by leveraging the Oklab color space. Using Oklab here over more advanced models such as CAM16 is useful because working out the math becomes a lot simpler. It also means that it won’t be a full color model able to adapt to different viewing conditions, but that is probably also desirable here since it is more practical.

> For consistency with the naming of Oklab, these new color spaces will be called Okhsl and Okhsv. The parameters will also be referred to as hhh, sss and lll and hhh, sss and vvv respectively. Those names are a bit confusing but I think making the new spaces easy to adopt for someone used to HSL and HSV is more important than trying to establish new names.

Reply:
HSL and HSV have distortions.

![image](https://user-images.githubusercontent.com/32321/175615329-56fa3c36-c6a8-42d2-b966-a81c09e0be0f.png)

![image](https://user-images.githubusercontent.com/32321/175615338-203e1cd8-7da0-4366-ab1e-08d7f933d6a7.png)


OK Color attempts to decrease the distortions.

![image](https://user-images.githubusercontent.com/32321/175615473-4c04afd7-3bbd-4b5b-b387-f7db2579c087.png)

![image](https://user-images.githubusercontent.com/32321/175615480-2af44dc2-2bc6-496e-beb5-5549a6e096a9.png)

Reply:
So I guess some brief extra description + link to the page (?) would be enough.